projects/adrv902x/src/hal/no_os_platform.c

```diff
@@ -0,0 +1,863 @@
+#ifdef PLATFORM_MB
+		.type = SPI_PL,
+#else
+		.type = SPI_PS,
+#endif
+		.flags = 0
+	};
+
+	/* sysref req GPIO configuration */
+	gip_gpio_reset_n.number = ADRV9025_RESET_B;
+	gip_gpio_reset_n.extra = &gip_extra;
+	gip_gpio_reset_n.platform_ops = &xil_gpio_ops;
+	ret = no_os_gpio_get(&phal->gpio_reset_n, &gip_gpio_reset_n);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_direction_output(phal->gpio_reset_n, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	struct no_os_spi_init_param sip = {
+		.device_id = SPI_DEVICE_ID,
+		.max_speed_hz = 10000000u,
+		.mode = NO_OS_SPI_MODE_0,
+		.chip_select = ADRV9025_CS,
+		.platform_ops = &xil_spi_ops,
+		.extra = &sip_extra
+	};
+
+	ret = no_os_spi_init(&phal->spi, &sip);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_set_value(phal->gpio_reset_n, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * \brief Gracefully shuts down the the hardware closing any open resources
+ *        such as log files, I2C, SPI, GPIO drivers, timer resources, etc.
+ *
+ * \param devHalCfg Pointer to device instance specific platform settings
+ *
+ * \retval ADI_HAL_OK Function completed successfully, no action required
+ * \retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ */
+int32_t no_os_HwClose(void *devHalCfg)
+{
+	int32_t ret;
+	struct adrv9025_hal_cfg *phal = (struct adrv9025_hal_cfg *)devHalCfg;
+	ret = no_os_gpio_remove(phal->gpio_reset_n);
+	if (ret)
+		return ret;
+
+	ret = no_os_spi_remove(phal->spi);
+	if (ret)
+		return ret;
+
+	return ADI_HAL_OK;
+}
+
+/**
+ * \brief This function control a BBIC GPIO pin that connects to the reset pin
+ *        of each device.
+ *
+ *  This function is called by each device API giving access to the Reset pin
+ *  connected to each device.
+ *
+ * \param devHalCfg Pointer to device instance specific platform settings
+ * \param pinLevel The desired pin logic level 0=low, 1=high to set the GPIO pin to.
+ *
+ * \retval ADI_HAL_OK Function completed successfully, no action required
+ * \retval ADI_HAL_NULL_PTR The function has been called with a null pointer
+ */
+int32_t no_os_HwReset(void *devHalCfg, uint8_t pinLevel)
+{
+	struct adrv9025_hal_cfg *phal = (struct adrv9025_hal_cfg *)devHalCfg;
+
+	if (devHalCfg == NULL) {
+		return ADI_HAL_NULL_PTR;
+	}
+
+	no_os_gpio_set_value(phal->gpio_reset_n, pinLevel);
+
+	return ADI_HAL_OK;
+}
+
+int32_t no_os_image_page_get(void *devHalCfg, const char *ImagePath,
+			     uint32_t pageIndex, uint32_t pageSize, uint8_t *rdBuff)
+{
+	unsigned char *bin;
+	if (!strcmp(ImagePath, "ADRV9025_FW.bin")) {
+		if ((pageIndex * pageSize) > sizeof(ADRV9025_FW_bin))
+			return -EINVAL;
+
+		bin = ADRV9025_FW_bin;
+	} else if (!strcmp(ImagePath, "ADRV9025_DPDCORE_FW.bin")) {
+		if ((pageIndex * pageSize) > sizeof(ADRV9025_DPDCORE_FW_bin))
+			return -EINVAL;
+		bin = ADRV9025_DPDCORE_FW_bin;
+	} else if (!strcmp(ImagePath,
+			   "stream_image_6E3E00EFB74FE7D465FA88A171B81B8F.bin")) {
```

Comment:
maybe we should avoid the hash from the name in order to avoid too many code changes when updating this file (this will be updated pretty often!)

Reply:
I created and used a define for it. This way, when a new stream is provided, only the define, the new file and the inclusion of the new file must be updated.